### PR TITLE
Add Google Analytics events for newsletter button clicks

### PR DESCRIPTION
**Pull request by DennisTsiang**
Add GA to the
- Newsletter subscribe button in the homepage
- Newsletter subscribe button in the footer

### CODE REVIEW

File: source/_newsletter.erb [hunk 1]
```diff
@@ -2,6 +2,6 @@
   <div>
     <div style="position:absolute;left:-5000px;"><input type="text" name="b_831facdc5caf30c092463f1c0_e924413a0b" tabindex="-1" value=""></div>
     <input class="<%= block_class %>__email" type="email" placeholder="Your email address" value="" name="EMAIL">
-    <input class="<%= block_class %>__button" type="submit" value="Subscribe" name="subscribe">
+    <input class="<%= block_class %>__button" type="submit" value="Subscribe" name="subscribe" onclick="<%= onclick_func %>" >
```
Comment (neilvanbeinum) [inline source/_newsletter.erb]: Nitpick: I think we can go with `onclick_function` here for a little readability tweak.

**Review by neilvanbeinum — CHANGES_REQUESTED** [review]

File: source/_newsletter.erb (hunk 1 above)
Comment (DennisTsiang) [reply source/_newsletter.erb]: Sure, I'll amend the commit

File: source/index.erb [hunk 2]
```diff
@@ -234,7 +234,10 @@ blog_title: "What we have been blogging about"
...
 
 <script>
-function sendToGaHomePageServiceClickEvent(serviceName) {
-  ga('send', 'event', 'Service', 'Click', serviceName, '1');
-}
+  function sendToGaHomePageServiceClickEvent(serviceName) {
+    ga('send', 'event', 'Service', 'Click', serviceName, '1');
+  }
+  function sendToGaHomePageSubscribeClickEvent() {
+        ga('send', 'event', 'Button', 'Click', 'Homepage newsletter subscribe', '1');
```
Comment (neilvanbeinum) [inline source/index.erb]: Sorry to be a nitpicker but can we make the indentation consistent here

Comment (DennisTsiang) [reply source/index.erb]: Good spot

**Review by neilvanbeinum — APPROVED** [review]

**Pull request merged**